sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -207,6 +207,25 @@ public void finish(@Nullable SpanStatus status) {
     this.finishStatus = FinishStatus.finishing(status);
     if (!root.isFinished() && (!waitForChildren || hasAllChildrenFinished())) {
       root.finish(finishStatus.spanStatus);
+
+      // finish unfinished children
+      Date finishTimestamp = root.getTimestamp();
```

Comment:
```suggestion
      final Date finishTimestamp = root.getTimestamp();
```

Reply:
It can't be final as it may be reassigned in line 221.